File: ios/MMCoach/MMCoach/Models/CorrectedDictationSegment.swift
//
//  CorrectedDictationSegment.swift
//  MMCoach
//
//  Result of BackendService.correctDictation, mirroring mmCorrectDictation's
//  response shape (see backend/cloud/schemas/aiResponseSchemas.js).
//

import Foundation

struct CorrectedDictationSegment: Decodable {
    let correctedSegment: String
    let changes: [Change]

    struct Change: Decodable {
        let original: String
        let corrected: String
    }
}

File: android/app/src/main/java/dev/benderapps/mmcoach/models/CorrectedDictationSegment.kt
package dev.benderapps.mmcoach.models

/** Result of BackendService.correctDictation, mirroring mmCorrectDictation's response shape. */
data class CorrectedDictationSegment(
    val correctedSegment: String,
    val changes: List<Change>,
) {
    data class Change(
        val original: String,
        val corrected: String,
    )
}
